cognite/client/_api_client.py

```diff
@@ -545,8 +566,11 @@ def _list(
             advanced_filter=advanced_filter,
             api_subversion=api_subversion,
         ):
-            items.extend(cast(T_CogniteResourceList, resource_list).data)
-        return list_cls(items, cognite_client=self._cognite_client)
+            if items is None:
+                items = cast(T_CogniteResourceList, resource_list)
+            else:
+                items.data.extend(cast(T_CogniteResourceList, resource_list).data)
+        return items if items is not None else list_cls([], cognite_client=self._cognite_client)
```

Comment:
I don't understand what's going on here? You change `items` initial value from empty list to `None` in order to...?

Reply:
Originally items was a regular list which was then passed into the list_cls. 

This means we lose extra information stored in the list, namely, typing.